pkg/scheduler/objects/allocation_ask_test.go

```diff
@@ -155,3 +155,39 @@ func TestGetRequiredNode(t *testing.T) {
 	ask = NewAllocationAsk(siAsk)
 	assert.Equal(t, ask.GetRequiredNode(), "NodeName", "required node should be NodeName")
 }
+
+func TestSetPriority(t *testing.T) {
+	var tests = []struct {
+		testMessage string
+		in          int32
+		out         int32
+	}{
+		{"no priority setting", -1, 0},
+		{"set default priority value", 0, 0},
+		{"set priority", 2, 2},
+	}
+
+	for _, tt := range tests {
+		res := resources.NewResourceFromMap(map[string]resources.Quantity{"first": 10})
+		siAsk := &si.AllocationAsk{
+			AllocationKey:  "ask-1",
+			ApplicationID:  "app-1",
+			MaxAllocations: 1,
+			ResourceAsk:    res.ToProto(),
+		}
+		t.Run(tt.testMessage, func(t *testing.T) {
+			ask := NewAllocationAsk(siAsk)
+			if tt.in >= 0 {
+				priority := si.Priority_PriorityValue{PriorityValue: tt.in}
+				siAsk.Priority = &si.Priority{
+					Priority: &priority,
+				}
+				ask.setPriority(siAsk.Priority.GetPriorityValue())
```

Comment:
I guess it's better to do this in a different way because this is not how we usually assign the priority value.
can we assign the value directly in `AllocationAsk`?

```
siAsk := &si.AllocationAsk{
  AllocationKey:  "ask-1",
  ApplicationID:  "app-1",
  MaxAllocations: 1,
  Priority:   xxxx,
  ResourceAsk:    res.ToProto(),
}
```

Reply:
Hi @yangwwei , thank for review.
Priority is usually assigned to si.AllocationAsk and handled via NewAllocationAsk function. 
I will additionally add this example in UT.
Test two ways which are NewAllocationAsk and SetPriority function to set priority.

Reply:
I add two situation in this commit.
1.Set priority via "NewAllocationAsk" function
2.Priority initialized by "NewAllocationAsk"  is  assigned new value by SetPriority.